clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -3619,4 +3619,14 @@ protected boolean isTransientRecordBufferUsed() {
   protected void setPartitionConsumptionState(int partition, PartitionConsumptionState pcs) {
     partitionConsumptionStateMap.put(partition, pcs);
   }
+
+  public void blockOnDrainerCapacity() {
```

Comment:
Nit: better naming could be maybeBlockOnDrainerCapacity as most likely this won't be blocked until extreme case.
Feel free to continue if there is no other comments as I understand this is a tactical fix and we would like to merge it asap.

Reply:
fixed